### PR TITLE
[feature] [postgresql] Add support for setting the password_encryption

### PR DESCRIPTION
**Pull request by nunofernandes**
### Description of the change

Add support for setting the `password_encryption` in the postgresql configuration file

### Benefits

On some applications where libpq 10 isn't available, we can't set the `password_encryption` to `md5`. As the default has changed to `scram-sha-256` in postgresql 14 and without a possibility to change that setting in the container, we can't use this container. This patch adds that feature.

### Possible drawbacks

n/a

### Applicable issues

- n/a

### Additional information

n/a

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for creating this PR. Please, note it is needed to sign the commits. Your signature certifies that you wrote the patch or otherwise have the right to contribute the material. You can find more info about this topic, in [the “Sign your work” section from the contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

**Comment by nunofernandes** [comment]
Signing was just done. Pls let me know if that works..

File: bitnami/postgresql/11/debian-11/rootfs/opt/bitnami/scripts/libpostgresql.sh [hunk 1]
```diff
@@ -620,6 +620,9 @@ postgresql_initialize() {
     # Configure port
     postgresql_set_property "port" "$POSTGRESQL_PORT_NUMBER"
     is_empty_value "$POSTGRESQL_DEFAULT_TOAST_COMPRESSION" || postgresql_set_property "default_toast_compression" "$POSTGRESQL_DEFAULT_TOAST_COMPRESSION"
+    if [[ -n "$POSTGRESQL_PASSWORD_ENCRYPTION" ]]; then
+        postgresql_set_property "password_encryption" "$POSTGRESQL_PASSWORD_ENCRYPTION"
+    fi
```
Comment (FraPazGal) [inline bitnami/postgresql/11/debian-11/rootfs/opt/bitnami/scripts/libpostgresql.sh:625]: Following the same syntax used by `$POSTGRESQL_DEFAULT_TOAST_COMPRESSION`, what do you think about changing this to:

```suggestion
    is_empty_value "$POSTGRESQL_PASSWORD_ENCRYPTION" || postgresql_set_property "password_encryption" "$POSTGRESQL_PASSWORD_ENCRYPTION"
```

Besides that, could you please add `$POSTGRESQL_PASSWORD_ENCRYPTION` to the `postgresql-env.sh` files? You can use [this PR](https://github.com/bitnami/containers/pull/38520/files#diff-d95a9000323413aebbbf2f79e3842d1d1239e8d4347b6502deac418e8af088c9) as an example.

**Review by FraPazGal — COMMENTED** [review]
Hi @nunofernandes, I have added a comment as a review. Please check it whenever you can!

File: bitnami/postgresql/11/debian-11/rootfs/opt/bitnami/scripts/libpostgresql.sh (hunk 1 above)
Comment (nunofernandes) [reply bitnami/postgresql/11/debian-11/rootfs/opt/bitnami/scripts/libpostgresql.sh]: Just did that.. Pls let me know if that works.. :pray:

**Comment by FraPazGal** [comment]
The PR looks good @nunofernandes, let me work on the needed internal changes and I'll come back to merge this. Thanks for the contribution!

**Review by FraPazGal — APPROVED** [review]

**Comment by nunofernandes** [comment]
Hi... Thanks again...

Can you help me understand what these changes will land on the docker image (namely bitnami/postgresql:14.8.0)?

**Comment by carrodher** [comment]
The latest version of the container image, for instance, `14.8.0-debian-11-r20` or any of the rolling tags (`14`, `14-debian-11`, ` 14.8.0`) pointing to this immutable tag should already contain the changes.

**Pull request merged**